src/Coordination/KeeperServer.cpp

```diff
@@ -282,11 +379,14 @@ void KeeperServer::putLocalReadRequest(const KeeperStorage::RequestForSession &
 
 RaftAppendResult KeeperServer::putRequestBatch(const KeeperStorage::RequestsForSessions & requests_for_sessions)
 {
-
     std::vector<nuraft::ptr<nuraft::buffer>> entries;
     for (const auto & [session_id, time, request] : requests_for_sessions)
         entries.push_back(getZooKeeperLogEntry(session_id, time, request));
 
+    std::lock_guard lock{server_mutex};
+    if (is_recovering)
+        return nullptr;
+
     return raft_instance->append_entries(entries);
```

Comment:
We need to do it with lock (server_mutex), right?

Reply:
yes, there is a lock taken 4 lines above?